packages/yew-macro/src/typed_vdom/mod.rs

```diff
@@ -0,0 +1,94 @@
+use proc_macro2::{Ident, TokenStream};
+use quote::{format_ident, quote};
+use syn::ext::IdentExt;
+use syn::parse::{Parse, ParseStream};
+use syn::{parse_quote, LitStr, Token, Type};
+
+pub mod generate_element;
+mod globals;
+pub use globals::*;
+
+#[derive(Clone)]
+pub struct AttributePropDefinition {
+    name: Ident,
+    ty: Type,
+}
+
+impl Parse for AttributePropDefinition {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        let name = Ident::parse_any(input)?;
+        let _separator = input.parse::<Token![:]>();
+        let ty = input.parse()?;
+        Ok(Self { name, ty })
+    }
+}
+
+impl AttributePropDefinition {
+    pub fn new(name: Ident, ty: Type) -> Self {
+        Self { name, ty }
+    }
+    fn build_fields(&self) -> TokenStream {
+        let AttributePropDefinition { ty, .. } = self;
+        let name = self.name();
+        quote! {
+            #[prop_or_default]
+            pub #name: ::std::option::Option::<#ty>,
+        }
+    }
+
+    fn name(&self) -> Ident {
+        format_ident!("r#{}", self.name)
+    }
+
+    fn build_if_lets(&self) -> TokenStream {
+        let ident = self.name();
+        let name = self.name.to_string().replace('_', "-");
+        let name = LitStr::new(&name, self.name.span());
+        quote! {
+            if let Some(val) = self.#ident {
```

Comment:
Not that it really matters here, but just `Some` isn't hygienic.

Reply:
It actually matters, just not now. I thought there would be some unhygienic code somewhere but didn't quite know where

Reply:
I absolutely agree that it matters, I just didn't want to come off as too pedantic :)